packages/typegpu/src/tgsl/generationHelpers.ts

```diff
@@ -519,19 +519,26 @@ export function convertToCommonType(
   ctx: GenerationCtx,
   values: Snippet[],
   restrictTo?: AnyData[],
+  verbose = true,
 ): Snippet[] | undefined {
   const types = values.map((value) => value.dataType);
 
   if (types.some((type) => type === UnknownData)) {
     return undefined;
   }
 
+  if (verbose && Array.isArray(restrictTo) && restrictTo.length === 0) {
```

Comment:
Maybe we should show it only in DEV mode? 👀 
In this way, the warnings code is going to be stripped in production builds
```suggestion
  if (DEV && verbose && Array.isArray(restrictTo) && restrictTo.length === 0) {
```